File: Low Level Design (LLD)/DesignPatterns/2. Behavariol/Flyweight/Java/Circle.java
public class Circle implements Shape {
    private String color; // Intrinsic state (shared)

    // Extrinsic state will be passed via draw method
    public Circle(String color) {
        this.color = color;
        System.out.println("Creating circle of color: " + color);
    }

    @Override
    public void draw(int x, int y, int radius) {
        System.out.println("Circle drawn [Color: " + color + ", X: " + x + ", Y: " + y + ", Radius: " + radius + "]\n");
    }
}

File: Low Level Design (LLD)/DesignPatterns/2. Behavariol/Flyweight/Cpp/src/Circle.cpp
#include <iostream>
#include "../include/Circle.h"

Circle::Circle(const std::string& color) : color(color) {
    std::cout << "Creating circle of color: " << color << std::endl;
}

void Circle::draw(int x, int y, int radius) {
    std::cout << "Circle drawn [Color: " << color 
              << ", X: " << x 
              << ", Y: " << y 
              << ", Radius: " << radius 
              << "]" << std::endl << std::endl;
}
